tests/form_params_test.go

```diff
@@ -0,0 +1,385 @@
+// ignored
+func TestFormParametersBreakUrlEncoding(t *testing.T) {
+	postReq := []byte("POST / HTTP/1.1\r\n" +
+		"Host: localhost:8080\r\n" +
+		"Content-Type: application/x-www-form-urlencoded\r\n" +
+		"Content-Length: 10\r\n" +
+		"\r\n" +
+		"veggie=%sc\r\n" +
+		"\r\n")
+	resp, err := requesttesting.MakeRequest(context.Background(), postReq, func(req *http.Request) {
+		if val := req.Form["veggie"]; len(req.Form["veggie"]) != 0 {
+			t.Errorf("req.Form[\"veggie\"]: got %s, want %s.", val, "")
+		}
+	})
+	if err != nil {
+		t.Fatalf("MakeRequest(): got err %v, want nil", err)
+	}
+	if !bytes.HasPrefix(resp, []byte(statusOKReq)) {
+		t.Errorf("response status: got %s, want %s", resp, statusOKReq)
+	}
+}
+
+func TestBasicMultipartForm(t *testing.T) {
+	reqBody := "--123\r\n" +
+		"Content-Disposition: form-data; name=\"foo\"\r\n" +
+		"\r\n" +
+		"bar\r\n" +
+		"--123--\r\n"
+	postReq := "POST / HTTP/1.1\r\n" +
+		"Host: localhost:8080\r\n" +
+		"Content-Type: multipart/form-data; boundary=\"123\"\r\n" +
+		"Content-Length: " + strconv.Itoa(len(reqBody)) + "\r\n" +
+		"\r\n" +
+		reqBody + "\r\n" +
+		"\r\n"
+	resp, err := requesttesting.MakeRequest(context.Background(), []byte(postReq), func(req *http.Request) {
+		e := req.ParseMultipartForm(1024)
+		if e != nil {
+			t.Fatalf("ParseMultipartForm: want nil, got %v", e)
+		}
+
+		if formVal := req.Form["foo"][0]; formVal != "bar" {
+			t.Errorf("FormValue: got %s, want bar", formVal)
+		}
+	})
+	if err != nil {
+		t.Fatalf("MakeRequest(): got err %v, want nil", err)
+	}
+	if !bytes.HasPrefix(resp, []byte(statusOKReq)) {
+		t.Errorf("response status: got %s, want %s", resp, statusOKReq)
+	}
+}
+
+// Test behaviour of multipart/form-data request passed with no Content-Length.
+// In this case, the body will be ignored and ParseMultiPartForm()å parsing will fail
+// with NextPart:EOF.  This is rather inconsistent with
+// application/x-www-form-urlencoded where a  missing Content-Length will
+// return 400 rather than 200.
+func TestMultipartFormNoContentLength(t *testing.T) {
+	// Skipping to ensure GitHub checks are not failing. This test is made
+	// to fail intentionally to pinpoint behaviour we regard as inconsistent
+	t.Skip()
+	reqBody := "--123\r\n" +
+		"Content-Disposition: form-data; name=\"foo\"\r\n" +
+		"\r\n" +
+		"bar\r\n" +
+		"--123--\r\n"
+	postReq := "POST / HTTP/1.1\r\n" +
+		"Host: localhost:8080\r\n" +
+		"Content-Type: multipart/form-data; boundary=\"123\"\r\n" +
+		"\r\n" +
+		reqBody +
+		"\r\n"
+	resp, err := requesttesting.MakeRequest(context.Background(), []byte(postReq), func(req *http.Request) {
+		t.Error("MakeRequest(): expected handler not to be called")
+	})
+
+	if err != nil {
+		t.Fatalf("MakeRequest(): got err %v, want nil", err)
+	}
+	if !bytes.HasPrefix(resp, []byte(statusBadReq)) {
+		t.Errorf("response status: got %s, want %s", resp, statusBadReq)
+	}
+}
+
+// Test behaviour of multipart/form-data request passed with smaller
```

Comment:
Make the test comment reflect what's the desired state and make the `t.Skip()` comment explain what's going wrong. This way, if Go's library gets fixed, you just need to remove the `t.Skip()` and accompanying comment.

Reply:
Resolved in [a523171](url)